pkg/descheduler/controllers/migration/filter.go

```diff
@@ -317,6 +333,20 @@ func (r *Reconciler) filterMaxMigratingOrUnavailablePerWorkload(pod *corev1.Pod)
 		}
 	}
 
+	podJobName := getPodMigratingJobName(pod)
+	podOwnJob := &sev1alpha1.PodMigrationJob{}
+	if podJobName != "" {
+		err := r.Client.Get(context.TODO(), client.ObjectKey{Name: podJobName}, podOwnJob)
```

Comment:
Already supports indexing PodMigrationJob by Pod UID. You can look at [pkg/descheduler/fieldindex/register.go#L71](https://github.com/koordinator-sh/koordinator/blob/main/pkg/descheduler/fieldindex/register.go#L71)